api/src/database/get-ast-from-query/lib/convert-wildcards.ts

```diff
@@ -30,7 +30,13 @@ export async function convertWildcards(options: ConvertWildcardsOptions, context
 			{
 				collection: options.parentCollection,
 				action: 'read',
-				accountability: options.accountability,
+				accountability: {
+					app: options.accountability.app,
+					role: options.accountability.role,
+					roles: options.accountability.roles,
+					user: options.accountability.user,
+					ip: options.accountability.ip ?? null,
```

Comment:
I wonder if it makes sense to introduce a `pick` argument to the `withCache` function, so the decision which parts of their arguments are cache-key-relevant can be made in that unit where the function lives and doesn't have to be made all over the code base. This current way seems very error prone to me, since one could easily forget to first of all pick the relevant arguments or forget to update all function signatures and also a bit confusing, took me a hot minute to verify that `admin` is indeed not needed in this function call.

What do you think?